internal/resources/ephemeral/ephemeral_resources.go

```diff
@@ -140,8 +140,11 @@ func (r *Resources) Close(ctx context.Context) tfdiags.Diagnostics {
 	r.mu.Lock()
 	defer r.mu.Unlock()
 
-	// We might be closing due to a context cancellation, but we still need
-	// to be able to make non-canceled Close requests.
+	// We might be closing due to a context cancellation, but we still need to
+	// be able to make non-canceled Close requests.
+	//
+	// TODO: if we're going to ignore the cancellation to ensure that Close is
```

Comment:
I think a timeout (or a force cancel?) would make sense

Reply:
I'll have to think on that, since anytime you pick a timeout it's going to be wrong for some situation. This is already in a path that shouldn't be happening under normal circumstances though, so maybe it doesn't really matter so much.